src/directives/offense/poisonRoom.ts

```diff
@@ -1,169 +1,188 @@
+		}
+	}
+	spawnMoarOverlords() {
+		this.overlords.claim = new ClaimingOverlord(this);
+		this.overlords.roomPoisoner = new RoomPoisonerOverlord(this);
+	}
+	init() {
+		this.alert(`Poisoning Room ${this.pos.roomName}`);
+
+		if(!(this.room && this.room.controller)) {
+			return;
+		}
+
+		// suspend room immediately once claimed to avoid default room actions
+		if(this.room.controller.level == 1) {
+			this.suspendColony(this.room.name);
+		}
+
+		if (Game.time % RUN_TIMER == 0 || this.room.constructionSites.length == 0) {
+			// capture variables
+			this.walkableSourcePosisions	 = _.filter(_.flatten(_.map(this.room.sources,
+													s => s.pos.neighbors)),pos => pos.isWalkable(true));
+			this.walkableControllerPosisions =  _.filter(this.room.controller!.pos.neighbors,
+													pos => pos.isWalkable(true));
+			
+			// reverse reservation if needed
+			if(this.room.controller.reservation && this.room.controller.reservation.ticksToEnd > 500) {
+				DirectiveControllerAttack.createIfNotPresent(this.room.controller.pos,'room');
+			}
+
+			// send fighters if needed
+			if(this.room && this.room.playerHostiles.length > 0) {
+				DirectiveOutpostDefense.createIfNotPresent(this.room.controller.pos,'room');
+			}
+		}
+	}
+	private suspendColony(roomName: string): string {
+		if (Overmind.colonies[roomName]) {
+			const colonyMemory = Memory.colonies[roomName] as ColonyMemory | undefined;
+			if (colonyMemory) {
+				colonyMemory.suspend = true;
+				Overmind.shouldBuild = true;
+				return `Colony ${roomName} suspended.`;
+			} else {
+				return `No colony memory for ${roomName}!`;
+			}
+		} else {
+			return `Colony ${roomName} is not a valid colony!`;
+		}
+	}
+
+	private poisonActions() {
+		if(!(this.room && this.room.controller && this.room.controller.level == 2 && 
+			 this.walkableControllerPosisions && this.walkableControllerPosisions)) {
+			return;
+		}
+
+		// Poison Controller First
+		if(this.walkableControllerPosisions.length > 0) {
+			_.first(this.walkableControllerPosisions).createConstructionSite(STRUCTURE_WALL);
+			return;
+		} 
+
+		// Poison Sources Next
+		// Trick: Csite are not walkable, wait for poisoner to carry energy before blocking Sources
+		// Trick: Remove sources Csites if poisoner need to harvest, i.e carrying 0 energy
+		if (this.overlords.roomPoisoner.roomPoisoners.length &&
+			this.overlords.roomPoisoner.roomPoisoners[0].carry.energy > 0) {
+				if(this.walkableSourcePosisions.length) {
+					_.first(this.walkableSourcePosisions).createConstructionSite(STRUCTURE_WALL);
+				}
+		} else {
+				_.forEach(this.room.constructionSites, csite => {csite.remove();} );
+		}
+		
+	}
+	private isPoisoned(): boolean {
+		let result = false;
+		if (this.room && this.room.controller!.level > 1) {
+			result = !!this.walkableSourcePosisions && !this.walkableSourcePosisions.length &&
+					 !!this.walkableControllerPosisions && !this.walkableControllerPosisions.length;
```

Comment:
Can this terminate early if there is just unwalkable construction sites next to them?

Reply:
good point, but it does not.
it looks like isWalkable ignores wall csites. otherwise my testing would have never worked and the rooms would get unclaimed on the last csite before getting built.